### ISSUE TITLE
Cross-browser compatibility in one built extension?

### ISSUE THREAD

**Issue by matatk** [matatk/landmarks#59]
Stumbled upon this [excellent article in Smashing mag on making cross-browser extensions](https://www.smashingmagazine.com/2017/04/browser-extension-edge-chrome-firefox-opera-brave-vivaldi/).  From that article, it looks like adding this code may enable me to do away with the Edge API bridge and also have it still work across Firefox and Chrome.

```js
window.browser = (function () {
  return window.msBrowser ||
    window.browser ||
    window.chrome;
})();
```

Will need to look into one thing, though: I gather [the `browser.` namespace on Firefox supports—and even *requires*—the use of promises rather than callbacks](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Chrome_incompatibilities), so some of the API calls may not work correctly there.  They do have a [polyfill for running on Chrome](https://github.com/mozilla/webextension-polyfill).  If only one bit of code needs to be built, that would simplify things a lot. Possibly worth seeing where Opera and Edge are with respect to supporting/requiring promises on their `browser.` namespaces first.

There's [a related mailing list thread](https://mail.mozilla.org/pipermail/dev-addons/2016-November/002101.html).

**Comment by matatk** [comment]
The code works in Firefox, Chrome and Opera. Whilst it's not needed in Firefox, it doesn't break anything, and not including it would actually make the manifests more complex, so will leave as-is for now.  Need to try it in Edge—if it works there, with no need for the API bridges, then it's definitley worth keeping.

Different browsers use different image sizes, and the build system is already set up to provide versions of the extension that cater to these, so at the moment it seems OTT to return to just one identical build for all browsers, though this is something that could be very nice in future, especially if SVG icons get supported.

**Comment by matatk** [comment]
Merged by #66
